ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -253,10 +259,49 @@ export default class TestDataModal extends React.Component {
     }));
 
     getTestData(displayedTestParams);
-    this.setState({ selectedTests: [], filterText: '' });
+    this.setState({
+      selectedTests: [],
+      selectedUnits: new Set(),
+      filterText: '',
+    });
     this.closeModal();
   };
 
+  selectableTestClassName = test => {
+    return this.hasDifferentUnit(test) ? 'bg-warning' : '';
+  };
+
+  selectableTestTitle = test => {
+    if (this.hasDifferentUnit(test)) {
+      return `Warning: ${this.getOriginalTestName(
+        test,
+      )} has a different measurement unit (${test.measurementUnit}) `;
+    }
+    return this.getOriginalTestName(test);
+  };
+
+  hasDifferentUnit = test => {
+    const { plottedUnits } = this.props;
+    const { selectedUnits } = this.state;
+    const unit = test.measurementUnit || null;
+
+    const differentThanPlottedUnits =
+      plottedUnits.size && !plottedUnits.has(unit);
+    const selectedUnitTypesMismatch = selectedUnits.size >= 2;
+    const differentThanSelectedUnits =
+      selectedUnits.size && !selectedUnits.has(unit);
+
+    return (
+      differentThanPlottedUnits ||
+      selectedUnitTypesMismatch ||
+      differentThanSelectedUnits
+    );
+  };
+
+  extractUniqueUnits(tests) {
+    return new Set(tests.map(aTest => aTest.measurementUnit || null));
```

Comment:
I think this will also return a `null` value in the returned array.

Reply:
I've addressed this also.